lib/managers/connection_check_status_ manager.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+import 'package:api/models/app/managers/connection_status.dart';
+
+class ConnectionCheckStatusManager implements ConnectionStatus {
+  static final ConnectionCheckStatusManager _connectionStatusSingleton =
+      ConnectionCheckStatusManager._internal();
+  ConnectionCheckStatusManager._internal();
+
+  factory ConnectionCheckStatusManager() => _connectionStatusSingleton;
+
+  bool _hasConnection = false;
+
+  StreamController connectionChangeController =
+      StreamController<bool>.broadcast();
+
+  final Connectivity _connectivity = Connectivity();
+
+  @override
+  void initialize() => _connectivity.onConnectivityChanged
+      .asBroadcastStream()
+      .listen(getConnectionChange);
+
+  void getConnectionChange(ConnectivityResult result) =>
+      hasInternetConnection();
+
+  @override
+  Stream get connectionChangeStream => connectionChangeController.stream;
+
+  void dispose() => connectionChangeController.close();
+
+  @override
+  Future<bool> hasInternetConnection() async {
```

Comment:
Maybe change this method to only return the last state of the connection of the "connectionChangeStream" with a bool return. 

And rename the current method like 'initializeConnectionStatusChangeStream'  
🤔